src/core/ext/transport/chttp2/transport/hpack_encoder.cc

```diff
@@ -180,14 +180,12 @@ static void evict_entry(grpc_chttp2_hpack_compressor *c) {
 
 /* add an element to the decoder table */
 static void add_elem(grpc_exec_ctx *exec_ctx, grpc_chttp2_hpack_compressor *c,
```

Comment:
could we split this into add_elem() and add_key(), having add_elem() just call add_key()?

Reply:
... if they have the same signature, we could eliminate the two bools in emit_indexed and just store a function pointer instead

Reply:
Yes. I did not do it because they will need to share the code to evict table. I have factored that to a separate function and added two proxy functions to share the call. I have changed to use function pointers in places (maybe too much). The microbm looks ok.